packages/block-library/src/classic/edit.js

```diff
@@ -84,9 +84,14 @@ export default class ClassicEdit extends Component {
 		}
 
 		editor.on( 'blur', () => {
+			const bookmark = editor.selection.getBookmark( 2, true );
+
 			setAttributes( {
 				content: editor.getContent(),
 			} );
+
+			editor.once( 'focus', () => editor.selection.moveToBookmark( bookmark ) );
+
```

Comment:
Question: What happens if we select some text, click outside of the classic block and then click somewhere in the classic block. Won't this create issues and try to reselect the previous selection?

Reply:
Yes, it will restore the selection. This is the proper behavior, needed when replacing content. 

To test:
1. Type something in a Classic Block.
2. Select part of it.
3. Click to open the media modal.
4. Select an image and insert it.

The selected text is properly replaced by the image.

Reply:
I think it feels like a bug if you move the cursor elsewhere, say you type in another block and then you want to click on a precise position of the classic block to write there again.

Reply:
Not sure what we can do about it. :)

Reply:
I think we should reset the bookmark if the block becomes unselected `isSelected` => false

Reply:
That's not a bad idea :)

Reply:
Hm, a better way of clearing the selection seems to be to do it on `mousedown` inside the editor div? We are worried only about restoring the selection after the block was clicked. 

Not sure what else may trigger the block to become "unselected", perhaps keyboard shortcut? That may happen while the block is under a modal.